File: 201902658/11047.py
if __name__ == "__main__":
    N, K = map(int, input().split())
    coins = []
    
    for _ in range(N):
        coins.append(int(input()))
        
    count = 0
    coins.sort(reverse = True)
    for coin in coins:
        count += K // coin
        K %= coin
    print(count)

File: 201902646/11047.cpp
#include <iostream>
#include <string>
int coin[11];

int main() {
    int N, K;
    int count = 0;
    std::cin >> N >> K;

    int num;
    for(int i=1; i<=N; i++) {
        std::cin >> coin[i];
    }
    
    int A = N;

    while(K) {
        while(K - coin[A] >= 0) {
            K -= coin[A];
            count++;
        }
        A--;
    }
    std::cout << count;
}
